src/librustc_mir/borrow_check/mod.rs

```diff
@@ -444,6 +444,9 @@ fn downgrade_if_error(diag: &mut Diagnostic) {
             "this represents potential undefined behavior in your code and \
             this warning will become a hard error in the future",
         );
+        diag.note(
```

Comment:
You can chain things here from the first `.warn` to the last `.note` since `.note` and the other methods returns `&mut Self`.